File: 프로그래머스/lv2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    que1 = deque(queue1)
    que2 = deque(queue2)
    sum1 = sum(que1)
    sum2 = sum(que2)
    tot = sum1 + sum2
    
    if tot % 2 == 1:
        return -1
    
    for _ in range(300000):
        if sum1 > sum2:
            tmp = que1.popleft()
            que2.append(tmp)
            sum1 -= tmp
            sum2 += tmp
            answer += 1
        elif sum1 < sum2:
            tmp = que2.popleft()
            que1.append(tmp)
            sum1 += tmp
            sum2 -= tmp
            answer += 1
        else:
            return answer
            
    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = -1;

        long sum1 = 0;
        Queue<Integer> q1 = new LinkedList<>();
        for (int e : queue1) {
            sum1 += e;
            q1.add(e);
        }

        long sum2 = 0;
        Queue<Integer> q2 = new LinkedList<>();
        for (int e : queue2) {
            sum2 += e;
            q2.add(e);
        }

        if ((sum1 + sum2) % 2 != 0) {
            return -1;
        }

        int len = queue1.length;

        int cnt = 0;
        while (true) {
            if (cnt > 3 * len) {
                break;
            }
            if (sum1 == sum2) {
                answer = cnt;
                break;
            } else if (sum1 > sum2) {
                int e = q1.poll();
                q2.add(e);
                sum1 -= e;
                sum2 += e;
            } else {
                int e = q2.poll();
                q1.add(e);
                sum1 += e;
                sum2 -= e;
            }
            cnt++;
        }

        return answer;
    }
}